packages/x-date-pickers/src/PickersTextField/PickersFilledInput/PickersFilledInput.tsx

```diff
@@ -51,97 +51,135 @@ const PickersFilledInputRoot = styled(PickersInputBaseRoot, {
     [`&.${pickersFilledInputClasses.disabled}`]: {
       backgroundColor: theme.vars ? theme.vars.palette.FilledInput.disabledBg : disabledBackground,
     },
-    ...(!ownerState.disableUnderline && {
-      '&::after': {
-        borderBottom: `2px solid ${
-          (theme.vars || theme).palette[ownerState.color || 'primary']?.main
-        }`,
-        left: 0,
-        bottom: 0,
-        // Doing the other way around crash on IE11 "''" https://github.com/cssinjs/jss/issues/242
-        content: '""',
-        position: 'absolute',
-        right: 0,
-        transform: 'scaleX(0)',
-        transition: theme.transitions.create('transform', {
-          duration: theme.transitions.duration.shorter,
-          easing: theme.transitions.easing.easeOut,
-        }),
-        pointerEvents: 'none', // Transparent to the hover style.
-      },
-      [`&.${pickersFilledInputClasses.focused}:after`]: {
-        // translateX(0) is a workaround for Safari transform scale bug
-        // See https://github.com/mui/material-ui/issues/31766
-        transform: 'scaleX(1) translateX(0)',
-      },
-      [`&.${pickersFilledInputClasses.error}`]: {
-        '&:before, &:after': {
-          borderBottomColor: (theme.vars || theme).palette.error.main,
+    variants: [
+      ...Object.keys((theme.vars ?? theme).palette)
+        // @ts-ignore
+        .filter((key) => (theme.vars ?? theme).palette[key].main)
+        .map((color) => ({
+          props: { color },
+          style: {
+            '&::after': {
+              // @ts-ignore
+              borderBottom: `2px solid ${(theme.vars || theme).palette[color]?.main}`,
+            },
+          },
+        })),
```

Comment:
This looks broken. 🙈 
It should only apply when `disableUnderline={false}`.

### Before
![Screenshot 2024-03-07 at 17 01 06](https://github.com/mui/mui-x/assets/4941090/8ffedb26-7f21-4922-82ee-e418ab8e3e3b)

### After
![Screenshot 2024-03-07 at 17 01 02](https://github.com/mui/mui-x/assets/4941090/42a19083-bc21-4fd1-8909-cd9d09a192c0)

Reply:
Nice catch, the `ownerState` received is the one passed by `PickersInputBase`
I'll see how to fix this

Reply:
Fixed